tests/test_bigbio.py

```diff
@@ -0,0 +1,660 @@
+         (4) test_schema: confirms big-bio keys present
+         (5) test_are_ids_globally_unique: Checks if all examples have a unique identifier
+
+         # KB-Specific tests
+         (6) test_do_all_referenced_ids_exist: Check if any sub-key (ex: entities/events etc.) have referenced keys
+         (7) test_passages_offsets: Check if text matches offsets in passages
+         (8) test_entities_offsets: Check if text matches offsets in entities
+         (9) test_events_offsets: Check if text matches offsets in events
+        (10) test_coref_ids: Check if text matches offsets in coreferences
+
+        """  # noqa
+        # self.test_name()
+        self.setUp()
+        # self.test_are_ids_globally_unique()
+
+        # KB-specific unit-tests
+        for task in self._SUPPORTED_TASKS:
+            self.test_schema(task)
+
+            mapped_task = _SCHEMA_MAP[_TASK_MAP[task.lower()]]
+            self.print_statistics(mapped_task)
+
+            # if _TASK_MAP[task.lower()] == "kb":
+            # self.test_do_all_referenced_ids_exist()
+            # self.test_passages_offsets()
+            # self.test_entities_offsets()
+            # self.test_events_offsets()
+            # self.test_coref_ids()
+
+    def setUp(self) -> None:
+        """Load original and big-bio schema views"""
+        log.info("Checking if data loads")
+        # TODO: Enforce if source is relevant
+        self.dataset_source = datasets.load_dataset(
+            self.PATH,
+            name="source",
+            data_dir=self.DATA_DIR,
+            use_auth_token=self.USE_AUTH_TOKEN,
+        )
+
+        self.dataset_bigbio = datasets.load_dataset(
+            self.PATH,
+            name=self.VIEW,
+            data_dir=self.DATA_DIR,
+            use_auth_token=self.USE_AUTH_TOKEN,
+        )
+
+        # Get task type of the dataset
+        log.info("Checking _SUPPORTED_TASKS present.")
+        self._SUPPORTED_TASKS = importlib.import_module(
+            "biomeddatasets." + self.NAME + "." + self.NAME
+        )._SUPPORTED_TASKS
+
+    def print_statistics(self, schema: Features):
+        """
+        Gets sample statistics, for each split and sample of the number of features in the schema present; only works for the big-bio schema.
+
+        :param schema_type: Type of schema to reference features from
+        """  # noqa
+        log.info("Gathering schema statistics")
+        for split_name, split in self.dataset_bigbio.items():
+            print(split_name)
+            print("=" * 10)
+
+            counter = defaultdict(int)
+            for example in split:
+                for feature_name, feature in schema.items():
+                    if example.get(feature_name, None) is not None:
+                        if isinstance(feature, datasets.Value) or isinstance(
+                            feature_name, dict
+                        ):
+                            if example[feature_name]:
```

Comment:
is the `isinstance(feature_name, dict)` correct above? i'm not sure a dict can be used as a key in `example`.

Reply:
@leonweber and sam wrote this function; i think the point is iterable not necessarily dict)